Makefile

```diff
@@ -146,11 +146,11 @@ else
 endif
 
 build-docs:
-	cargo doc --release $(compiler_features) --document-private-items --no-deps --workspace
+	cargo doc --release --no-deps --manifest-path lib/api/Cargo.toml
```

Comment:
Unsure about this change here, might want to talk with @nlewycky about it.

My understanding of our custom build of the docs was to be different from what docs.rs does, but these settings seem like it'd provide the same (or maybe a worse?) experience than docs.rs.

The `--no-deps` in particular, I think means a lot of types that could be linked to in the generated docs won't be linked to.